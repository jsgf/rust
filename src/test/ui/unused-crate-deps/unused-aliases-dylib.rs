// Warn about unused aliased for the crate

// edition:2018
// check-pass
// aux-crate:bar=bar.rs
// aux-crate:barbar=bar.rs
// ignore-tidy-linelength

#![warn(unused_crate_deps)]
//~^ WARNING External crate `barbar` unused in `unused_aliases_dylib`. Remove the dependency or add `use barbar as _;`.

use bar as _;

fn main() {}
