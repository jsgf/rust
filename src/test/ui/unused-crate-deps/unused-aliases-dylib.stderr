warning: External crate `barbar` unused in `unused_aliases_dylib`. Remove the dependency or add `use barbar as _;`.
  --> $DIR/unused-aliases-dylib.rs:9:1
   |
LL | #![warn(unused_crate_deps)]
   | ^
   |
note: the lint level is defined here
  --> $DIR/unused-aliases-dylib.rs:9:9
   |
LL | #![warn(unused_crate_deps)]
   |         ^^^^^^^^^^^^^^^^^

